File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    rank=7
    zeronum=0
    answer=[]
    for i in lottos:
        if i in win_nums:
            rank-=1
        elif i==0:
            zeronum+=1
    if zeronum==0 and rank==7:
        answer.append(6)
        answer.append(6)
    else:
        available_rank=rank-zeronum
        answer.append(available_rank)

        if rank==7:
            rank=6
        answer.append(rank)
    return answer

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];// 최고 순위, 최저 순위
        int rank=7;
        int zeronum=0;
        ArrayList<Integer> wn = new ArrayList<>();
        for(int num:win_nums){
            wn.add(num);
        }
        
        
        for(int i : lottos){
            if(wn.contains(i)){
                rank-=1;
            }else if(i==0){
                zeronum+=1;
            }
        }
        if(zeronum==0 && rank==7){
            answer[0]=6;
            answer[1]=6;
        }else{
            int available_rank= rank-zeronum;
            answer[0]=available_rank;
            if(rank==7){
                rank=6;
            }
            answer[1]=rank;
            
        }
        return answer;
    }
}